lib/routing.js

```diff
@@ -62,24 +66,18 @@ export const deserializeState = serializedState => {
   return JSON.parse(decodeURIComponent(stateString))
 }
 
-const getQueryStringObject = query => {
-  if (query.state) {
-    return deserializeState(query.state)
+export const getRouteState = router => {
+  const { asPath = '' } = router
+  const { query, pathname } = url.parse(asPath, true)
+  const queryState = getQueryStringState(query)
+  const path = escapeHtml(pathname.split('/').pop())
+  // TODO fix this hack
+  const parameter = path.length >= 19 && path.indexOf('.') === -1 && path
+
+  return {
+    parameter,
+    queryState
   }
-
-  const state = mapper.map(mappings, query)
-  deserializeCode(state)
-
-  Object.keys(state).forEach(key => {
-    if (state[key] === '') state[key] = undefined
-  })
-
-  return state
-}
-
-export const getQueryStringState = query => {
-  const queryParams = getQueryStringObject(query)
-  return Object.keys(queryParams).length ? queryParams : {}
 }
 
 export const updateQueryString = (router, state) => {
```

Comment:
@raboid now that we added `getRouteState`, what do you think about renaming this `updateRouteState`?

Reply:
Sounds like a good name to me!